openfe_gromacs/protocols/gromacs_md/md_settings.py

```diff
@@ -0,0 +1,668 @@
+# For details, see https://github.com/OpenFreeEnergy/openfe-gromacs
+
+"""Settings class for plain MD Protocols using Gromacs + OpenMMTools
+
+This module implements the settings necessary to run MD simulations using
+:class:`openfe_gromacs.protocols.gromacs_md.md_methods.py`
+
+"""
+from typing import Literal, Optional
+
+from gufe.settings import OpenMMSystemGeneratorFFSettings, SettingsBaseModel
+from openfe.protocols.openmm_utils.omm_settings import (
+    IntegratorSettings,
+    OpenFFPartialChargeSettings,
+    OpenMMEngineSettings,
+    OpenMMSolvationSettings,
+    Settings,
+)
+from openff.models.types import FloatQuantity
+from openff.units import unit
+
+try:
+    from pydantic.v1 import validator
+except ImportError:
+    from pydantic import validator  # type: ignore[assignment]
+
+
+class SimulationSettings(SettingsBaseModel):
+    """
+    Settings for simulations in Gromacs.
+    """
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    # # # Run control # # #
+    integrator: Literal["md", "sd", "steep"] = "sd"
+    """
+    MD integrators and other algorithms (steep for energy minimization).
+    Allowed values are:
+    'md': a leap-frog integrator
+    'sd': A leap-frog stochastic dynamics integrator. Note that when using this
+          integrator the parameters tcoupl and nsttcouple are ignored.
+    'steep': A steepest descent algorithm for energy minimization.
+    """
+    dt: FloatQuantity["picosecond"] = 0.002 * unit.picosecond
+    """
+    Time step for integration (only makes sense for time-based integrators).
+    Default 0.002 * unit.picosecond
+    """
+    nsteps: int = 0
+    """
+    Maximum number of steps to integrate or minimize, -1 is no maximum
+    Default 0
+    """
+    mass_repartition_factor: int = 1
+    """
+    Scales the masses of the lightest atoms in the system by this factor to
+    the mass mMin. All atoms with a mass lower than mMin also have their mass
+    set to that mMin. Default 1 (no mass scaling)
+    """
+
+    # # # Langevin dynamics # # #
+    ld_seed: int = -1
+    """
+    Integer used to initialize random generator for thermal noise for
+    stochastic and Brownian dynamics. When ld-seed is set to -1,
+    a pseudo random seed is used. Default -1.
+    """
+
+    # # # Neighbor searching # # #
+    cutoff_scheme: Literal["verlet"] = "verlet"
+    """
+    Only allowed option:
+    'verlet': Generate a pair list with buffering. The buffer size is
+              automatically set based on verlet-buffer-tolerance, unless this
+              is set to -1, in which case rlist will be used.
+    """
+    nstlist: int = 10
+    """
+    Frequency to update the neighbor list. When dynamics and
+    verlet-buffer-tolerance set, nstlist is actually a minimum value and
+    gmx mdrun might increase it, unless it is set to 1.
+    If set to zero the neighbor list is only constructed once and never updated.
+    Default 10.
+    """
+    pbc: Literal["xyz", "no", "xy"] = "xyz"
```

Comment:
I can't remember re: minimization, but is there a use case we can see right now where pbc != xyz?

Reply:
I can't think of a use case right now either, so maybe I will move this to be not user interfacing.

Reply:
Moved this to the PRE_DEFINED_SETTINGS in PR #7